src/lib.rs

```diff
@@ -32,14 +33,17 @@ pub fn inline_lib() -> RichTerm {
     let lib_ncl = include_bytes!(concat!(env!("OUT_DIR"), "/main.ncl"));
     let lib_ncl = String::from_utf8_lossy(lib_ncl);
 
-    let mut cache = Cache::new(ErrorTolerance::Strict);
+    let mut cache = CacheHub::new();
     let parser = TermParser::new();
-    let file_id = cache.add_string(
+    let file_id = cache.sources.add_string(
         SourcePath::Generated("main.ncl".to_owned()),
         lib_ncl.to_string(),
     );
-    let lexer = Lexer::new(cache.source(file_id));
-    parser.parse_strict(file_id, lexer).unwrap()
+    let lexer = Lexer::new(cache.sources.source(file_id));
+    parser
+        .parse_strict(&AstAlloc::new(), file_id, lexer)
+        .unwrap()
+        .to_mainline()
```

Comment:
So for now, we just convert to the old representation and keep working on that. It's ok for the time being, but I suppose we'd like to eventually only operate on the new AST, which is much more compact and nicer. Maybe we should open an issue or something like that to remember?